extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/responseheader/ResponseHeaderTest.java

```diff
@@ -0,0 +1,111 @@
+package io.quarkus.resteasy.reactive.server.test.responseheader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.jboss.resteasy.reactive.Header;
+import org.jboss.resteasy.reactive.ResponseHeader;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.restassured.http.Headers;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+public class ResponseHeaderTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
+
+    @Test
+    public void should_return_added_headers_uni() {
+        Map<String, String> expectedHeaders = Map.of(
+                "Access-Control-Allow-Origin", "*",
+                "Keep-Alive", "timeout=5, max=997");
+        RestAssured
+                .given()
+                .get("/test/multi")
+                .then()
+                .statusCode(200)
+                .headers(expectedHeaders);
+    }
+
+    @Test
+    public void should_return_added_headers_multi() {
+        Map<String, String> expectedHeaders = Map.of(
+                "Access-Control-Allow-Origin", "*",
+                "Keep-Alive", "timeout=5, max=997");
+        RestAssured
+                .given()
+                .get("/test/multi")
+                .then()
+                .statusCode(200)
+                .headers(expectedHeaders);
+    }
+
+    @Test
+    public void should_throw_exception_without_headers_uni() {
```

Comment:
Can you also add a test for the case where a resource method returns a string but throws an exception?
The same should also be done for the status test

Reply:
Don't quite understand you here, current implementation supports only Uni and Multi therefore if the resource method will return simple String current logic will not work

Reply:
Okay, then there was misunderstanding of my initial comment.

In my initial I meant that the annotations should support all cases - and the original tests only tested for the non-reactive case.
We need to support and test all of them.